code/game/objects/items/weaponry.dm

```diff
@@ -117,7 +117,7 @@
 
 /obj/item/claymore/highlander/attack(mob/living/target, mob/living/user)
 	. = ..()
-	if(target && target.stat == DEAD && target.mind && target.mind.special_role == "highlander")
+	if(target && target.stat == DEAD && target.mind && target.mind.special_role == "highlander" && !isbrain(target))

```

Comment:
I think it'd be better for dust() to return TRUE/FALSE based on result, and check that.